satpy/readers/__init__.py

```diff
@@ -401,6 +424,30 @@ def available_readers(as_dict=False, yaml_loader=UnsafeLoader):
     return readers
 
 
+def _assign_loadables_to_reader_name(fs, loadables, reader_instance_name, reader_files):
+    """Assign files (loadables) to reader names.
+
+    If a :class:`fsspec.spec.AbstractFileSystem` is passed the file pathes are converted to
+    :class:`FSFile` objects.
+
+    Args:
+        fs (:class:`fsspec.spec.AbstractFileSystem`): fsspec filesystem instance.
+        loadables (list[str]): List of filepathes.
+        reader_instance_name (str): Reader name.
+        reader_files (dict): Dictionary with list of filepathes associated with each reader name.
+
+    Returns:
+        dict: Dictionary with list of filepathes associated with each reader name.
+
+    """
+    if fsspec and fs is not None and isinstance(fs, fsspec.spec.AbstractFileSystem):
```

Comment:
Do we really need to check `fs is not None`? shouldn't the isinstance catch it?

Reply:
Yes I think so because it is possible that `fsspec` is installed but `fs` is not an `fsspec` filesystem.